protocol/client/messenger_v2.go

```diff
@@ -67,85 +81,77 @@ func (m *MessengerV2) Start() error {
 			if exist {
 				return fmt.Errorf("multiple public chats with same topic: %s", contacts[i].Topic)
 			}
-			stream := NewStream(context.Background(), options, m.proto, NewPublicHandler(contacts[i], m.db))
-			err := stream.Start()
-			if err != nil {
+
+			stream := NewStream(m.proto, StreamHandlerForContact(contacts[i], m.db))
+			if err := stream.Start(context.Background(), options); err != nil {
 				return errors.Wrap(err, "unable to start stream")
 			}
 			m.public[contacts[i].Topic] = stream
 		}
 	}
-	log.Printf("[INFO] request messages from mail sever")
+
+	log.Printf("[Messenger::Start] request messages from mail sever")
+
 	return m.RequestAll(context.Background(), true)
 }
 
 func (m *MessengerV2) Join(ctx context.Context, c Contact) error {
 	m.mu.Lock()
 	defer m.mu.Unlock()
-	if c.Type == ContactPublicRoom {
-		return m.joinPublic(ctx, c)
+	if c.Type == ContactPublicKey {
```

Comment:
why?) you just converted it to 2nd option

Reply:
Because in `Start()` there was `if contacts[i].Type == ContactPublicKey {` as the first case and wanted to keep it consistent.